src/AutoVizuA11y.tsx

```diff
@@ -265,6 +291,18 @@ const AutoVizuA11y = ({
 		asyncEffect();
 	}, [chartRef]);
 
+	if (!isValidElement(shortcutGuide)) {
+		shortcutGuide = <ShortcutGuide closeShortcutGuide={closeShortcutGuide} />;
+	} else {
+		shortcutGuide = cloneElement(shortcutGuide, { closeShortcutGuide });
+	}
+
+	if (!isValidElement(shortcutGuide)) {
+		shortcutGuide = <ShortcutGuide closeShortcutGuide={closeShortcutGuide} />;
+	} else {
+		shortcutGuide = cloneElement(shortcutGuide, { closeShortcutGuide });
+	}
+
```

Comment:
It looks to me like the current approach is to import a component named <ShortcutGuide /> and expect a prop named `shortcutGuide`. The first being the `native` component, and the second the `foreign`. We then check if the prop `shortcutGuide` is a valid react component and depending on the result, _reasign_ the prop to either the native or the foreign component. 
Reasigning prop values is a _bad practice_ in React, props should be _read only_ so the components are as pure as possible and so data won't change unexpectedly. While reading the code is also hard to understand what is the `native` component and what is the `foreign` component.
With this in mind a propose a few changes:

**Rename <ShorcutGuide />**
To better differentiate between the shortcut guide that is from the client application and the one we provide, I suggest we rename the component `ShortcutGuide` to `NativeShortcutGuide`. This allows us to keep the prop name `shortcutGuide` which I think is a good option for the library  client's devx. It also has the slight plus of indicating that there are other ways to render a shortcut guide.

**Delegate the choice to a separate component**
Instead of doing the logic of deciding what component to render inside the `AutuVizua11y` component, create a new component `ShorcutGuideContainer` to encapsulate this logic. This will also solve the prop reasign issue, improving performance and making the main component easier to understand.
```
*ShortcutGuideContainer.tsx*
const ShortcutGuideContainer = (shortcutGuide, nativeShortcutGuide, onCloseShortcutGuide) => {
    if(isValidElement(shortcutGuide)) {
        return cloneElement(shortcutGuide, { closeShortcutGuide })
    }

    return nativeShortcutGuide({onCloseShortcutGuide})
}

*AutovizuA11y.tsx*
...
{isVisibleShortcutGuide && <ShortcutGuideContainer {...apropriateProps}/>}
```